### PR TITLE
adding homework LINQ

### PR DESCRIPTION
**Pull request by KrisAramis**
please review my LINQ homework

### CODE REVIEW

File: Helpers/Helpers.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿namespace Kristina_Kulich__Application.Helpers;
+
+public static class Helpers
```
Comment (dnnyhlwdplc) [inline Helpers/Helpers.cs]: Helpers -> Helper

File: Helpers/Helpers.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+{
+    private static int iseven;
+    public static bool CheckIfIsDevidedWithoutRamainder(int number, int devider)
+    {
+        //if (number % devider == 0)
```
Comment (dnnyhlwdplc) [inline Helpers/Helpers.cs]: Please remove commented code!

File: Program.cs [hunk 3]
```diff
@@ -1,16 +1,161 @@
...
-     new Homework_1().Task_2();
-     new Homework_1().Task_3();
-     new Homework_1().Task_4();
-     new Homework_1().Task_5();
-     new Homework_1().Task_6();
-     }
+        #region Task1
+
+        int[] integers = new[] { 12, 8, 27, 78, 2, 3, 3, 4, 21, 1 };
+        List<int> divedsby3 = integers.Select(i => i).Where(i => (i % 3 == 0)).ToList();
```
Comment (dnnyhlwdplc) [inline Program.cs]: what is a purpose of Select(i => i)

Comment (dnnyhlwdplc) [inline Program.cs]: there should be query syntax

File: Program.cs [hunk 4]
```diff
@@ -1,16 +1,161 @@
...
+        foreach (var v in divedsby3)
+        {
+            Console.WriteLine("Devides by 3 without remainder: {0}", v);
+        }
+
+        #endregion
+
+        #region Task2
+
+        int[] integers2 = new[] { 15, -8, -7, -5, 6, 14, 13, -9, 11, -10 };
```
Comment (dnnyhlwdplc) [inline Program.cs:35]: why there are 2 duplicated pieces of code?

File: Program.cs [hunk 5]
```diff
@@ -1,16 +1,161 @@
...
+        }
+
+        #endregion
+
+        #region Task3
+
+        List<string> brands = new List<string> { "chanel", "dior", "gucci", "stella mccartni", "adidas" };
+        var filteredbrands = from b in brands
+            where Helpers.CheckIfIsDevidedWithoutRamainder(b.Length, 2) ==
+                  true // b.Length== Helpers.CheckIfIsDevidedWithoutRamainder(b.Length, 2)
```
Comment (dnnyhlwdplc) [inline Program.cs]: please remove commented code!

File: Program.cs [hunk 6]
```diff
@@ -1,16 +1,161 @@
...
+        {
+            Console.WriteLine("Brands have even length: {0}", v);
+        }
+
+        #endregion
+
+        #region Task4
+
+        List<string> cities = new List<string>() { "Berlin", "London", "Paris", "Bobruisk", "Tokyo" };
+        var citiesfiltered = cities.Select(i => i).Where(i => i.StartsWith("B") && i.EndsWith("k")).ToList();
```
Comment (dnnyhlwdplc) [inline Program.cs]: it is better to not hardcode values, use variables instead. Example var startsWithNumber = "B"

File: Program.cs [hunk 7]
```diff
@@ -1,16 +1,161 @@
...
+
+        foreach (var g in citiesfiltered)
+        {
+            Console.WriteLine("City of dream is {0}", g);
+        }
+
+        #endregion
+
+        #region Task5
+        //couldn't solve, here my drafts
```
Comment (dnnyhlwdplc) [inline Program.cs]: var words = exampleToList.Split(new char[] { ' ' }).ToList();

File: Program.cs [hunk 8]
```diff
@@ -1,16 +1,161 @@
...
+        //   IsLower(allstring,)
+
+        //       if()
+        // List<string> allcases = new List<string>() { "BERLIN", "london", "paris", "BOBRUISK" ,"TOKYO" };
+
+        #endregion
+
+        #region Task6 
+        //couldn't solve yet, here drafts
+        int[] integers3 = new[] { 15, 4, -7, 15, -7, 14, 13, 4, 11, 15 };
```
Comment (dnnyhlwdplc) [inline Program.cs:91]: integers3.Distinct.Reverse

File: Program.cs [hunk 9]
```diff
@@ -1,16 +1,161 @@
...
+        for (int i = 0; i < list.Count - 1; i++)
+        {
+            listlength +=list[0].Length;
+        }
+        return listlength;
+    }
+    public static bool IsTheNumberInMassive(int numb, int[] integers)
+    {
+        int userinput = 0;
+        int.TryParse(Console.ReadLine(),out userinput);
```
Comment (dnnyhlwdplc) [inline Program.cs]: suggestion: please do not read from console in your custom methods :)

File: Program.cs [hunk 10]
```diff
@@ -1,16 +1,161 @@
...
+    public static int GetListAllElementsHeight(List<string> list)
+    {
+        int listlength = 0;
+        for (int i = 0; i < list.Count - 1; i++)
+        {
+            listlength +=list[0].Length;
+        }
+        return listlength;
+    }
+    public static bool IsTheNumberInMassive(int numb, int[] integers)
```
Comment (dnnyhlwdplc) [inline Program.cs]: why do we need numb? it is not used in method

File: Program.cs [hunk 11]
```diff
@@ -1,16 +1,161 @@
...
+    {
+        int userinput = 0;
+        int.TryParse(Console.ReadLine(),out userinput);
+        bool isInArray = false;
+        var arrayMatch = from n in integers
```
Comment (dnnyhlwdplc) [inline Program.cs]: integers.Contains(userinput)

File: Program.cs [hunk 12]
```diff
@@ -1,16 +1,161 @@
...
+            where n == userinput
+            select n;
+        if (arrayMatch.Any())
```
Comment (dnnyhlwdplc) [inline Program.cs]: return arrayMatch.Any()

File: Program.cs [hunk 13]
```diff
@@ -1,16 +1,161 @@
...
+
+        #endregion
+    }
+
+    public static int GetListAllElementsHeight(List<string> list)
+    {
+        int listlength = 0;
+        for (int i = 0; i < list.Count - 1; i++)
+        {
+            listlength +=list[0].Length;
```
Comment (dnnyhlwdplc) [inline Program.cs]: list[i].Length;

File: Program.cs [hunk 14]
```diff
@@ -1,16 +1,161 @@
...
+        #region Task8
+
+        /*List<string> twocolor = new List<string>();
+
+        List<string> tricolor = new List<string>() { "#FFFFFF", "FF0000", "#FFFFFF" };*/
+        //var selectedcolors = (tricolor.Count > 5).select(tricolor.First, tricolor.Last(a=> i)).ToList();
+
+        /*var selectedcolors = from t in tricolor 
+        where GetListAllElementsHeight(tricolor) > 5
+        select (tricolor.SelectMany(tricolor.First(), tricolor.Last()));*/// not sure how to select 1st and 3rd elements
```
Comment (dnnyhlwdplc) [inline Program.cs]: tricolor.First(), tricolor.Last() -  yes, this is correct

File: CSharpLinq/Task_1.cs [hunk 15]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections;
+using System.Linq;
+
+namespace Kristina_Kulich__Application.CSharpLinq;
```
Comment (dnnyhlwdplc) [inline CSharpLinq/Task_1.cs]: please not push unused classes!

File: Program.cs [hunk 16]
```diff
@@ -1,16 +1,161 @@
...
+
+        //Console.WriteLine($"First and last colors are:{selectedcolors}");
+
+        #endregion
+
+        #region Task9
+
+        IList<Student> studentList = new List<Student>()
```
Comment (dnnyhlwdplc) [inline Program.cs:125]: Great job!

**Review by dnnyhlwdplc — CHANGES_REQUESTED** [review]

File: Helpers/Helpers.cs (hunk 1 above)
Comment (KrisAramis) [reply Helpers/Helpers.cs]: fixed

File: Helpers/Helpers.cs (hunk 2 above)
Comment (KrisAramis) [reply Helpers/Helpers.cs]: fixed

File: Program.cs (hunk 3 above)
Comment (KrisAramis) [reply Program.cs]: code is red underlined without it "Cannot resolve method 'Select()', candidates are:   System.Collections.Generic.IEnumerable<TResult> Select<int,TResult>(this System.Collections.Generic.IEnumerable<int>, System.Func<int,int,TResult>) (in class Enumerable)   System.Collections.Generic.IEnumerable<TResult> Select<int,TResult>(this System.Collections.Generic.IEnumerable<int>, System.Func<int,TResult>) (in class Enumerable)"

Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs (hunk 4 above)
Comment (KrisAramis) [reply Program.cs:35]: fixed

File: Program.cs (hunk 5 above)
Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs (hunk 6 above)
Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs (hunk 10 above)
Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs (hunk 11 above)
Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs (hunk 14 above)
Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs (hunk 16 above)
Comment (KrisAramis) [reply Program.cs:125]: Thanks!

File: Program.cs (hunk 12 above)
Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs (hunk 9 above)
Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs (hunk 14 above)
Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs (hunk 13 above)
Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs (hunk 12 above)
Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs (hunk 7 above)
Comment (KrisAramis) [reply Program.cs]: fixed

File: Program.cs [hunk 17]
```diff
@@ -1,16 +1,166 @@
...
+        foreach (var g in citiesfiltered)
+        {
+            Console.WriteLine("City of dream is {0}", g);
+        }
+
+        #endregion
+
+        #region Task5
+
+        //couldn't solve, here my drafts
```
Comment (dnnyhlwdplc) [inline Program.cs:75]: Please remove unused comments

File: Program.cs [hunk 18]
```diff
@@ -1,16 +1,166 @@
...
+        foreach (var s in uppareCaseWords)
+        {
+            Console.WriteLine($"Upper case words are: {s}");
+        }
+        
+        #endregion
+
+        #region Task6
+
+        //couldn't solve yet, here drafts
```
Comment (dnnyhlwdplc) [inline Program.cs:90]: same

File: Program.cs [hunk 19]
```diff
@@ -1,16 +1,166 @@
...
+        }
+
+        #endregion
+
+
+        #region Task7
+
+        int userinput = 0;
+        int.TryParse(Console.ReadLine(),out userinput);
+        bool checkIfinMassive = IsTheNumberInMassive(userinput,integers);
```
Comment (dnnyhlwdplc) [inline Program.cs:105]: Massive -> Array

File: Program.cs [hunk 20]
```diff
@@ -1,16 +1,166 @@
...
+
+        #endregion
+    }
+    
+    public static bool IsTheNumberInMassive(int userinput, int[] integers)
+    {
+        return integers.Contains(userinput);
+    }
 }
+
```
Comment (dnnyhlwdplc) [inline Program.cs]: remove extra spaces

**Review by dnnyhlwdplc — APPROVED** [review]

**Pull request merged**